modules/firefox/hm.nix

```diff
@@ -0,0 +1,29 @@
+{ config, lib, ... }:
+
+let
+  profileSettings = {
+    settings = {
+      "font.name.monospace.x-western" = config.stylix.fonts.monospace.name;
+      "font.name.sans-serif.x-western" = config.stylix.fonts.sansSerif.name;
+      "font.name.serif.x-western" = config.stylix.fonts.serif.name;
+    };
+  };
+  makeProfileSettingsPair = profileName:
+    lib.nameValuePair profileName profileSettings;
+in {
+  options.stylix.targets.firefox = {
+    enable =
+      config.lib.stylix.mkEnableTarget "Firefox" config.programs.firefox.enable;
+
+    profileNames = lib.mkOption {
+      description = "The Firefox profile names to apply styling on.";
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
```

Comment:
I have tried the following but I get an infinite recursion:

```nix
{
  default = lib.attrNames config.programs.firefox.profiles;
}
```

If you have a suggestion, I would be happy to try. Otherwise, I don't think I can do what you want.

Reply:
This makes sense, it would probably need support from upstream to add an option similar to `home-manager.sharedModules`